### PR TITLE
Implement `V8Function`

### PR DESCRIPTION
**Pull request by noituri**
This PR makes it possible to create and call JS functions. I intend to use this later on for retrieving canvas position

### CODE REVIEW

File: compositor_chromium/src/v8/context.rs [hunk 1]
```diff
@@ -33,6 +36,24 @@ impl V8Context {
             Ok(V8Value::from_raw(get_global(ctx)))
         }
     }
+
+    pub fn eval(&self, code: &str) -> Result<V8Value, V8ContextError> {
```
Comment (wkozyra95) [inline compositor_chromium/src/v8/context.rs:40]: based on offline discussion this is not needed, so I would remove it.

**Review by wkozyra95 — APPROVED** [review]

**Review by WojciechBarczynski — APPROVED** [review]

File: compositor_chromium/src/v8/function.rs [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        Self(Validated::new(inner))
+    }
+
+    pub fn call(
+        &self,
+        args: &[V8Value],
+        ctx_entered: &V8ContextEntered,
+    ) -> Result<V8Value, V8FunctionError> {
+        self.inner_call(None, args, ctx_entered)
+    }
```
Comment (jerzywilczek) [inline compositor_chromium/src/v8/function.rs:34]: If F panics, the unwinding panic will cross into C++ land, which is a big no-no. 
read [this](https://doc.rust-lang.org/std/panic/fn.catch_unwind.html) and catch the panic.

**Review by jerzywilczek — CHANGES_REQUESTED** [review]
:crab:

**Review by jerzywilczek — APPROVED** [review]
:rocket:

**Pull request merged**